mlfinlab/portfolio_optimization/hcaa.py

```diff
@@ -224,13 +232,130 @@ def _quasi_diagnalization(self, num_assets, curr_index):
 
         return (self._quasi_diagnalization(num_assets, left) + self._quasi_diagnalization(num_assets, right))
 
+    def _recursive_bisection(self,
+                             expected_asset_returns,
+                             asset_returns,
+                             covariance_matrix,
+                             assets,
+                             allocation_metric,
+                             optimal_num_clusters,
+                             confidence_level):
+        # pylint: disable=bad-continuation, too-many-locals
```

Comment:
Please don't ignore this: `disable=bad-continuation`.

I realise that before we had these methods that have one argument per line but we have since updated our convention. Please let it rather read: 

```
def _recursive_bisection(self, expected_asset_returns, asset_returns, covariance_matrix, assets,
allocation_metric, optimal_num_clusters, confidence_level):
```

Reply:
Also I certainly agree with the linter that there are `too-many-locals`.

The logic inside should be broken into smaller methods. This will also improve readability.

Reply:
Done